File: src/net/auoeke/cin/lexer/lexeme/MappingLexeme.java
package net.auoeke.cin.lexer.lexeme;

import net.auoeke.cin.lexer.error.SyntaxError;

public final class MappingLexeme extends Lexeme {
    public MappingLexeme(int line, int column, SyntaxError error) {
        super(line, column, error);
    }

    public MappingLexeme(int line, int column) {
        super(line, column);
    }

    @Override public Type type() {
        return Type.EQUALS;
    }

    @Override public String toString() {
        return "=";
    }
}

File: src/net/auoeke/cin/lexer/lexeme/MappingLexeme.kt
package net.auoeke.cin.lexer.lexeme

class MappingLexeme(line: Int, column: Int, override val type: Type) : Lexeme(line, column) {
    override fun toString(): String = when (type) {
        Type.EQUALS -> "="
        Type.COLON -> ":"
        else -> throw IllegalStateException()
    }
}
